src/protocol.h

```diff
@@ -216,29 +237,44 @@ extern const char *FEEFILTER;
  * @since protocol version 70014 as described by BIP 152
  */
 extern const char *SENDCMPCT;
+constexpr uint8_t SENDCMPCT_SHORT_ID = 34;
 /**
  * Contains a CBlockHeaderAndShortTxIDs object - providing a header and
  * list of "short txids".
  * @since protocol version 70014 as described by BIP 152
  */
 extern const char *CMPCTBLOCK;
+constexpr uint8_t CMPCTBLOCK_SHORT_ID = 16;
 /**
  * Contains a BlockTransactionsRequest
  * Peer should respond with "blocktxn" message.
  * @since protocol version 70014 as described by BIP 152
  */
 extern const char *GETBLOCKTXN;
+constexpr uint8_t GETBLOCKTXN_SHORT_ID = 23;
 /**
  * Contains a BlockTransactions.
  * Sent in response to a "getblocktxn" message.
  * @since protocol version 70014 as described by BIP 152
  */
 extern const char *BLOCKTXN;
+constexpr uint8_t BLOCKTXN_SHORT_ID = 15;
 };
 
 /* Get a vector of all valid message types (see above) */
 const std::vector<std::string> &getAllNetMessageTypes();
 
+// Short Command IDs are a low bandwidth representations of a message type
+// The mapping is a peer to peer agreement
+
+// returns the short command ID for a command
+// returns 0 if no short command ID has been found
+uint8_t GetShortCommandIDFromCommand(const std::string cmd);
```

Comment:
caa6c58 can reference instead of copy?
```diff
-uint8_t GetShortCommandIDFromCommand(const std::string cmd);
+uint8_t GetShortCommandIDFromCommand(const std::string& cmd);
```

Reply:
Absolutely. Changed now.